blockchain/system/kip113.go

```diff
@@ -30,15 +32,50 @@ import (
 type BlsPublicKeyInfo struct {
 	PublicKey []byte
 	Pop       []byte
+	VerifyErr error // Nil if valid. Must check before use.
 }
 
-type (
-	BlsPublicKeyInfos map[common.Address]BlsPublicKeyInfo
-	AllocKip113Init   struct {
-		Infos BlsPublicKeyInfos
-		Owner common.Address
+func newBlsPublicKeyInfo(publicKey []byte, pop []byte) BlsPublicKeyInfo {
+	return BlsPublicKeyInfo{
+		PublicKey: publicKey,
+		Pop:       pop,
+		VerifyErr: verifyBlsPublicKeyInfo(publicKey, pop),
 	}
-)
+}
+
+func verifyBlsPublicKeyInfo(publicKey []byte, pop []byte) error {
+	pk, err := bls.PublicKeyFromBytes(publicKey)
+	if err != nil {
+		return err
+	}
+
+	sig, err := bls.SignatureFromBytes(pop)
+	if err != nil {
+		return err
+	}
+
+	if !bls.PopVerify(pk, sig) {
```

Comment:
System contract has pop validation, not yet been implemented. Is it not a duplicated?

Reply:
The KIP-113 spec suggests off-chain validation. https://github.com/klaytn/kips/blob/main/KIPs/kip-113.md. "Note The function is not able to verify the validity of the public key and the proof-of-possession due to the lack of EIP-2537. See validation for off-chain validation."
Considering that EIP-2537 isn't going to be added anytime soon, the off-chain validation is necessary.

Reply:
Got it. So, the [implementation mentioned](https://github.com/klaytn/system-contracts/blob/main/contracts/consensus/SimpleBlsRegistry.sol#L40) wouldn't be effective for now, correct?

Reply:
@hyunsooda No. I suspect we don't add EIP-2537 until Ethereum does, which has unclear timeline https://ethereum-magicians.org/t/eip-2537-bls12-precompile-discussion-thread/4187/48.